modules/local/cellrangerarc/mkref/main.nf

```diff
@@ -0,0 +1,39 @@
+process CELLRANGERARC_MKREF {
+    tag "$reference_config"
+    label 'process_medium'
+
+    container "nf-core/cellranger-arc:2.0.2"
+
+    // Exit if running this module with -profile conda / -profile mamba
+    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
+        exit 1, "CELLRANGERARC_COUNT module does not support Conda. Please use Docker / Singularity / Podman instead."
+    }
+
+    input:
+    path fasta
+    path gtf
+    path motifs
+    path reference_config
+    val reference_name
+
+    output:
+    path "${reference_name}", emit: reference
+    path "versions.yml"     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    cellranger-arc \\
+        mkref \\
+        --config=$reference_config \\
+        $args
```

Comment:
Not something I'll insist on, but I think it would be simpler and less confusing if the `generate_config.py` would be part of this process instead of a separate one. Here it seems that files get passed into the process that aren't used in the `script` section as it's not immediately clear what the `reference_config` is and that it contains the other filenames.

Reply:
I did it because I was unsure how much 10x will change about the gneration process of this reference config and the user could technically provide their own config file without using the generate config process. It just seemed cleaner to me that way. But I can change it if it too confusing.